Tasks/DownloadBuildArtifactsV0/main.ts

```diff
@@ -305,9 +307,52 @@ async function main(): Promise<void> {
                         var webProvider = new providers.WebProvider(itemsUrl, templatePath, variables, handler);
                         var fileSystemProvider = new providers.FilesystemProvider(downloadPath);
 
-                        downloadPromises.push(downloader.processItems(webProvider, fileSystemProvider, downloaderOptions).catch((reason) => {
-                            reject(reason);
-                        }));
+                        const downloadPromise = new Promise(async (downloadComplete, downloadFailed) => {
```

Comment:
Can we move it to separate function to avoid duplication?

Reply:
I reworked the architecture of the artifact download to avoid code duplication.